File: CS/WindowsApplication1/Form1.cs
using System;
using DevExpress.XtraEditors;

namespace Example
{
    public partial class Form1 : XtraForm
    {
        public Form1()
        {
            InitializeComponent();
            InitListBoxes();
        }

        private void InitListBoxes()
        {
            for (int i = 0; i < 10; i++)
            {
                customCheckedListBoxControl1.Items.Add(String.Format("Item {0}", i));
                customListBoxControl1.Items.Add(String.Format("Item {0}", i));
            }

            customCheckedListBoxControl1.ShiftItem += OnShiftItem;
            customListBoxControl1.ShiftItem += OnShiftItem;
        }

        private const int Offset = 20;
        private void OnShiftItem(object sender, ItemInfoEventArgs customEventArgs)
        {
            if (customEventArgs.ItemIndex % 2 == 1)
                customEventArgs.ItemOffset = Offset;
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports System
Imports DevExpress.XtraEditors

Namespace Example
	Partial Public Class Form1
		Inherits XtraForm

		Public Sub New()
			InitializeComponent()
			InitListBoxes()
		End Sub

		Private Sub InitListBoxes()
			For i As Integer = 0 To 9
				customCheckedListBoxControl1.Items.Add(String.Format("Item {0}", i))
				customListBoxControl1.Items.Add(String.Format("Item {0}", i))
			Next i

			AddHandler customCheckedListBoxControl1.ShiftItem, AddressOf OnShiftItem
			AddHandler customListBoxControl1.ShiftItem, AddressOf OnShiftItem
		End Sub

		Private Const Offset As Integer = 20
		Private Sub OnShiftItem(ByVal sender As Object, ByVal customEventArgs As ItemInfoEventArgs)
			If customEventArgs.ItemIndex Mod 2 = 1 Then
				customEventArgs.ItemOffset = Offset
			End If
		End Sub
	End Class
End Namespace